packages/mediapipe-core/lib/src/task_options.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:typed_data';
+import 'package:equatable/equatable.dart';
+import 'mediapipe_common_bindings.dart' as bindings;
+import 'ffi_utils.dart';
+
+class BaseOptions extends Equatable {
+  const BaseOptions({this.modelAssetBuffer, this.modelAssetPath})
+      : assert(
+          !(modelAssetBuffer == null && modelAssetPath == null),
+          'You must supply either `modelAssetBuffer` or `modelAssetPath`',
+        ),
+        assert(
+          !(modelAssetBuffer != null && modelAssetPath != null),
+          'You must only supply one of `modelAssetBuffer` and `modelAssetPath`',
+        );
+
+  /// The model asset file contents as bytes;
+  final Uint8List? modelAssetBuffer;
+
+  /// Path to the model asset file.
+  final String? modelAssetPath;
+
+  void toStruct(bindings.BaseOptions struct) {
+    if (modelAssetPath != null) {
+      struct.model_asset_path = prepareString(modelAssetPath!);
+    }
+    if (modelAssetBuffer != null) {
+      struct.model_asset_buffer = prepareUint8List(modelAssetBuffer!);
+    }
+  }
+
+  @override
+  List<Object?> get props => [modelAssetBuffer, modelAssetPath];
+}
+
+class ClassifierOptions extends Equatable {
+  const ClassifierOptions({
```

Comment:
Are there requirements that any of these values are set? What happens if they all are null?